process/peer/process.go

```diff
@@ -167,13 +167,18 @@ func (vs *validatorStatistics) saveUpdatesForNodesMap(
 	nodesMap map[uint32][][]byte,
 	peerType core.PeerType,
 ) error {
-	for shardID, pks := range nodesMap {
-		err := vs.saveUpdatesForList(pks, shardID, peerType)
+	for shardID := uint32(0); shardID < vs.shardCoordinator.NumberOfShards(); shardID++ {
```

Comment:
please add a TODO here to further investigates what is really going on. Someone else might pick up the task.

Reply:
done